kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs

```diff
@@ -279,6 +282,385 @@ impl<'tcx> GotocCtx<'tcx> {
         }
     }
 
+    fn codegen_rvalue_enum_single(
```

Comment:
Can you please add documentation to your functions? It really helps to review the logic.

Reply:
Done.